avocado/core/suite.py

```diff
@@ -141,6 +142,7 @@ def _from_config_with_resolver(cls, config, name=None):
             raise TestSuiteError(details)
 
         tasks = resolutions_to_tasks(resolutions, config)
+        tasks = create_requirements_tasks(tasks)
```

Comment:
Are you going to remove the line above? Are you going to introduce the config to this new method?

Because AFAICT, we use config to filter by tags, for instance. So I'm not sure how this is preserved.

Reply:
Maybe I should rename the `tasks` variables to something more meaningful. The thing here is, `create_requirements_tasks` needs the test tasks created to update the `prereqs` of each task and create the new requirements tasks based on the list of requirements of each test task.

While I was implementing, I thought the code would become more complex if I add the requirements resolution inside the `resolutions_to_tasks` function.

So, the code is basically extending the `tasks` list and updating the test tasks that already exist.

Would it make the code clear if I do:

```python
test_tasks = resolutions_to_tasks(resolutions, config)
all_tasks = create_requirements_tasks(test_tasks)
```

Or something like that?

Reply:
I think your new variable names are a good idea, they make your intent a lot clearer. One could realize it still if they spend more time on the code but this is a cognitive speedup.

I am not sure about all the information the config contains but specifically to the filters the current implementation should be fine because the config will be fully used to filter our the `test_tasks` and then the requirements only for those will be parsed and added to the overall list.